### ISSUE TITLE
ord exception in msgtypes

### ISSUE THREAD

**Issue by LucaCerina** [mclarkk/lifxlan#46]
Hi,
I want to notice that the line 388 in msgtypes.py gets faulty when the light label is not handled correctly.

Sometimes short labels have trailing \x00 in self.label and the 'join' used to generate the label crashes because 'ord(c)' expects a char while \x00 is considered an int

**Comment by jusmop** [comment]
I found this issue too. I replaced 'ord(c)' with 'c' and it worked for me, but I've not thoroughly tested it.

**Comment by jusmop** [comment]
I take that it back - replacing 'ord(c)' with 'c' is not the solution. Sorry for the noise.

**Comment by mclarkk** [comment]
I'm pretty sure what the problem was and I'm pretty sure I fixed it, but I'm not sure how to reproduce the problem on my end. Can you pull, rebuild, and test your code to confirm for me whether the fix worked?

In case you're interested: In `unpack.py` the labels were extracted with the following line:

`label = binascii.unhexlify("".join(["%2.2x" % (b & 0x000000ff) for b in struct.unpack("b"*32, payload_str[0:32])]))`

I confirmed that this could result in trailing 0x00 characters with the following:

```
>>> binascii.unhexlify("6369")
b'ci'
>>> binascii.unhexlify("636900")
b'ci\x00'
```

So I went through `unpack.py` and everywhere that `label` was unpacked I appended `.replace(b'\x00', b'')` resulting in the final horrific thunderline:

`label = binascii.unhexlify("".join(["%2.2x" % (b & 0x000000ff) for b in struct.unpack("b"*32, payload_str[0:32])])).replace(b'\x00', b'')`

**Comment by NicoTuxx** [comment]
Hi ! Firstly, thank you for your amazing work on lifx lan protocol :+1: !

Just here to say that I've got the same issue as @LucaCerina this day using hello_world.py.

I found that lifxlan was working perfectly installed and launched with python2.7, whereas python3.4 was having this issue.

The label in msgtypes.py is `<str>` for py2.7, but it is `<bytes>` for py3.4, which result to pass a `<str>` (c) to ord in py2.7 and it's good, but a `<int>` (c) to ord in py3.4 and there's the problem !

I've got a solution to fix this issue, by adding `.decode('utf-8')` after all the `.replace(b'\x00', b'')` to transform bytes in string for py3. I've tested this on py2.7 and py3.4 and it works for both.

You can see at my quick fix in this commit :
https://github.com/NicoTuxx/lifxlan/commit/27deeedc4d0e15f1daca4ae8ae68602509efbb50

I'm pretty new to python, so I'm pretty sure that I've got the origin of the issue, but probably not the prettiest way to fix it :|. If this is the good way, I'd be glad to open a PR :) !

**Comment by Kirillism** [comment]
@NicoTuxx me and a friend who had the same problem have tried your fix (Mac and Windows, both on Python 3.6) and it worked. We haven't tested it thoroughly, but so far so good. Thank you!

**Comment by mclarkk** [comment]
Added in the fix! Thanks @NicoTuxx!
